example/detox/e2e/init.js

```diff
@@ -33,59 +52,106 @@ beforeAll(async () => {
-    launchServer("Mockserver", mockserver, siteUrl, "../mocks");
+    launchServer("Mockserver", mockserver, siteUrl, { secure: false });
+}
+
+function launchSecureMockserver() {
+    launchServer(
+        "Secure Mockserver",
+        mockserver,
+        secureSiteUrl,
+        { secure: true },
+        secureServerOptions,
+        https
+    );
 }
 
 function launchWebSocketServer() {
     const port = webSocketSiteUrl.split(":")[2];
-    webSocketServer(port);
+    webSocketServer(port, { secure: false });
     console.log(`WebSocket Server listening at ${webSocketSiteUrl}`);
 }
 
-function launchServer(serverName, requestListener, url, directory = "") {
+function launchSecureWebSocketServer() {
+    const port = secureWebSocketSiteUrl.split(":")[2];
+    webSocketServer(port, { secure: true, serverOptions: secureServerOptions });
+    console.log(
+        `Secure WebSocket Server listening at ${secureWebSocketSiteUrl}`
+    );
+}
+
+function launchServer(
+    serverName,
+    requestListener,
+    url,
+    requestListenerParams = {},
+    serverOptions = {},
+    protocol = http
+) {
+    console.log(`Launching ${serverName} ...`);
     const port = url.split(":")[2];
-    const listeningMessage = `${serverName} listening at ${url}`;
-    const notListeningMessage = `${serverName} not listening at port ${port}! Launching ${serverName} (directory: ${directory}) ...`;
+    const listeningMessage = `${serverName} listening at ${url} with params (${JSON.stringify(
+        requestListenerParams
+    )})`;
 
-    // Check if server is listening
-    http.get(url, (res) => {
-        jestexpect(res.statusCode).toBe(200);
-        console.log(listeningMessage);
-    }).on("error", (e) => {
-        // Launch server if not listening
-        console.log(notListeningMessage);
-        const listener = directory
-            ? requestListener(directory)
-            : requestListener();
-        const server = http.createServer(listener).listen(port);
-        checkServerStatus(server, listeningMessage);
-    });
+    // Create server
+    const listener = requestListenerParams
+        ? requestListener(requestListenerParams)
+        : requestListener();
+    const server = protocol.createServer(serverOptions, listener).listen(port);
+
+    // Check server status
+    checkServerStatus(server, serverName, port, listeningMessage);
 }
 
-function checkServerStatus(server, listeningMessage) {
+function checkServerStatus(server, serverName, port, listeningMessage) {
     server.on("listening", () => {
         console.log(listeningMessage);
     });
-    server.on("error", (err) => {
-        console.log(err);
-        process.exit(1);
+    server.on("error", (error) => {
+        if (error.code === "EADDRINUSE") {
+            console.log(
+                `Another ${serverName} instance is already listening at ${port}`
+            );
+        } else {
+            console.log(error);
+            process.exit(1);
+        }
```

Comment:
There are now too many servers. Could it be grouped into logical types, like secure and non-secure server only? In short, can it be consolidated into 2 servers only?

Reply:
See comment https://github.com/mattermost/react-native-network-client/pull/57#issuecomment-815842308